src/js/background/messageHandler.js

```diff
@@ -273,7 +275,34 @@ const messageHandler = {
     }).catch((e) => {
       throw e;
     });
-  }
+  },
+
+  async maybePrepareSurveyAchievementOnUpdate(countOpened) {
+    if (countOpened < 10) {
+      return;
+    }
+
+    // Show the survey only for English locales (en or en-*).
+    const uiLang = browser.i18n.getUILanguage();
+    const lang = (uiLang || "").toLowerCase();
+    if (lang !== "en" && !lang.startsWith("en-")) {
+      return;
+    }
+
+    // Check if already scheduled in the past; if so, do not show again.
+    const achievementsStorage = await browser.storage.local.get({ achievements: [] });
+    const achievements = achievementsStorage.achievements;
```

Comment:
nit: I think this could be destructured _(something akin to this)_? 
```suggestion
    const { achievements } = await browser.storage.local.get({ achievements: [] });
```